test-scripts/get-descendants.lua

```diff
@@ -0,0 +1,21 @@
+local root = Instance.new("Folder")
+root.Name = "Root"
+
+local childA = Instance.new("Folder")
+childA.Name = "A"
+childA.Parent = root
+
+local grandchild = Instance.new("Folder")
+grandchild.Name = "Grandchild"
+grandchild.Parent = childA
+
+local childB = Instance.new("Folder")
+childB.Name = "B"
+childB.Parent = root
+
+local descendants = root:GetDescendants()
+
+assert(descendants[1] == childA, "`A` was not the first in descendants, instead it was " .. tostring(descendants[1]))
+assert(descendants[2] == childB, "`B` was not the first in descendants, instead it was " .. tostring(descendants[2]))
+assert(descendants[3] == grandchild, "`Grandchild` was not the first in descendants, instead it was " .. tostring(descendants[3]))
```

Comment:
This test would pass regardless of whether we used depth-first or breadth-first traversal. We need to!

Additionally, these error messages aren't quite right, presumably because they were copy-pasted! Why don't we write a table that contains our expected descendants and then write a loop to compare expected vs actual?